### PR TITLE
786- Job History Improvements

### LINKED ISSUE #786: Job History improvements
- In the error popup use a table style, not list style
- Use the actual date and time rather than time ago
- Duration - convert to human duration
- Add filters on job type, resource type, id, success
- Move sorting server side
- Convert from CamelCase to Camel Case in the job names

### LINKED ISSUE #1138: Close out 793
https://github.com/flanksource/flanksource-ui/pull/793

### PR DESCRIPTION
**Pull request by DesmondSanctity**
Fixes #786  and Fixes #1138

- [x] In the error popup use a table style, not list style
- [x] Use the actual date and time rather than time ago
- [x] Duration - convert to human duration
- [x] Add filters on job type, resource type, id, success
- [x] Move sorting server side
- [x] Convert from CamelCase to Camel Case in the job names

### CODE REVIEW

**Comment by moshloop** [comment]
@DesmondSanctity can you finish up the remaining items ?  If you have a look at the configs or incidents list page it should have the patterns that you can use.

**Comment by DesmondSanctity** [comment]
Yeah. I'm almost done with filters. The server side sorting how do I know the endpoints?

**Comment by moshloop** [comment]
@DesmondSanctity it is the same endpoint, just additional params that need to be added - check how it works for configs and incidents

**Comment by DesmondSanctity** [comment]
Okay got it.

**Comment by DesmondSanctity** [comment]
@moshloop This is done. Please check and drop feedbacks.

**Comment by moshloop** [comment]
@DesmondSanctity 
- I think there is an issue with duration - it might be assuming seconds instead of millisconds - we also probably want to use formatDuruation from utils to get values like '1h' or '1m23s'
- There is no filter for status
- The popup window should not have s/n - it should be just a single column
- The sort, should also be integrated into the table - see the config table

**Comment by moshloop** [comment]
@DesmondSanctity please only import date functions from utils.js, not from dayjs directly - The duration is wrong, most jobs return in milliseconds not seconds and it should display 25ms, or 1s or 2m1s

**Comment by DesmondSanctity** [comment]
> @DesmondSanctity please only import date functions from utils.js, not from dayjs directly - The duration is wrong, most jobs return in milliseconds not seconds and it should display 25ms, or 1s or 2m1s

Noted. I'll change it.

**Comment by DesmondSanctity** [comment]
@moshloop you can check now. Thanks

**Comment by DesmondSanctity** [comment]
@moshloop any other improvements for Job history??

**Comment by mainawycliffe** [comment]
@moshloop Do we want to create views for Job History Names and Resource IDs, the way we are currently fetching them will be hit by DB limits as the job history grows.

File: src/utils/date.ts [hunk 1]
```diff
@@ -116,6 +118,29 @@ export const formatTimeRange = (date: string | Date) => {
   });
 };
 
+export const formatDateToSeconds = (date: string | Date) =>
```
Comment (moshloop) [inline src/utils/date.ts:121]: Don't use a new function/format

for the duration.Humanize()

+ Customize durations across the board:

Use d vs day,

Comment (mainawycliffe) [reply src/utils/date.ts]: @moshloop is this what you had in mind?

**Comment by mainawycliffe** [comment]
@moshloop this should be good to go now

**Pull request merged**